swarm/network/stream/messages.go

```diff
@@ -80,12 +83,17 @@ func (p *Peer) handleRequestSubscription(ctx context.Context, req *RequestSubscr
 		// exchange between peers over p2p. Instead, error will be returned
 		// only if there is one from sending subscribe error message.
 		err = p.Send(ctx, SubscribeErrorMsg{
-			Error: err.Error(),
+			Server: true,
```

Comment:
It is actually the other way around, the server sends a `RequestSubscriptionMsg`, and the client runs `handleRequestSubscription`, which then sends in this case a `SubscribeErrorMsg` back to the server.

Reply:
Maybe we could rename the `Server` field to `SourceIsServer` or something like that.